diff --git SOIL2.c SOIL2.c
--- SOIL2.c
+++ SOIL2.c
@@ -365,7 +365,9 @@ unsigned int
 		const char *filename,
 		int force_channels,
 		unsigned int reuse_texture_ID,
-		unsigned int flags
+		unsigned int flags,
+		unsigned int * pWidth,
+		unsigned int * pHeight
 	)
 {
 	/*	variables	*/
@@ -426,6 +428,10 @@ unsigned int
 			reuse_texture_ID, flags,
 			GL_TEXTURE_2D, GL_TEXTURE_2D,
 			GL_MAX_TEXTURE_SIZE );
+
+	if ( pWidth != NULL ) *pWidth = width;
+	if ( pHeight != NULL ) *pHeight = height;
+
 	/*	and nuke the image data	*/
 	SOIL_free_image_data( img );
 	/*	and return the handle, such as it is	*/
diff --git SOIL2.h SOIL2.h
--- SOIL2.h
+++ SOIL2.h
@@ -166,7 +166,9 @@ unsigned int
 		const char *filename,
 		int force_channels,
 		unsigned int reuse_texture_ID,
-		unsigned int flags
+		unsigned int flags,
+		unsigned int * pWidth,
+		unsigned int * pHeight
 	);
 
 /**
